glean-core/android/src/test/java/mozilla/telemetry/glean/private/JweMetricTypeTest.kt

```diff
@@ -0,0 +1,141 @@
+        const val AUTH_TAG: String = "XFBoMYUZodetZdvTiFvSkQ"
+        const val JWE: String = "eyJhbGciOiJSU0EtT0FFUCIsImVuYyI6IkEyNTZHQ00ifQ.OKOawDo13gRp2ojaHV7LFpZcgV7T6DVZKTyKOMTYUmKoTCVJRgckCL9kiMT03JGeipsEdY3mx_etLbbWSrFr05kLzcSr4qKAq7YN7e9jwQRb23nfa6c9d-StnImGyFDbSv04uVuxIp5Zms1gNxKKK2Da14B8S4rzVRltdYwam_lDp5XnZAYpQdb76FdIKLaVmqgfwX7XWRxv2322i-vDxRfqNzo_tETKzpVLzfiwQyeyPGLBIO56YJ7eObdv0je81860ppamavo35UgoRdbYaBcoh9QcfylQr66oc6vFWXRcZ_ZT2LawVCWTIy3brGPi6UklfCpIMfIjf7iGdXKHzg.48V1_ALb6US04U3b.5eym8TW_c8SuK0ltJ3rpYIzOeDQz7TALvtu6UG9oMo4vpzs9tX_EFShS8iB7j6jiSdiwkIr3ajwQzaBtQD_A.XFBoMYUZodetZdvTiFvSkQ"
+        const val MINIMUM_JWE: String = "eyJhbGciOiJSU0EtT0FFUCIsImVuYyI6IkEyNTZHQ00ifQ...5eym8TW_c8SuK0ltJ3rpYIzOeDQz7TALvtu6UG9oMo4vpzs9tX_EFShS8iB7j6jiSdiwkIr3ajwQzaBtQD_A."
+    }
+
+    @get:Rule
+    val gleanRule = GleanTestRule(ApplicationProvider.getApplicationContext())
+
+    @Test
+    fun `The API saves to its storage engine`() {
+        // Define a 'jweMetric' jwe metric, which will be stored in "store1"
+        val jweMetric = JweMetricType(
+            disabled = false,
+            category = "telemetry",
+            lifetime = Lifetime.Application,
+            name = "jwe_metric",
+            sendInPings = listOf("store1")
+        )
+
+        // Record two JWEs of the same type, with a little delay.
+        jweMetric.set(HEADER, KEY, INIT_VECTOR, CIPHER_TEXT, AUTH_TAG)
+
+        // Check that data was properly recorded.
+        assertTrue(jweMetric.testHasValue())
+        assertEquals(JWE, jweMetric.testGetCompactRepresentation())
+
+        jweMetric.set(HEADER, "", "", CIPHER_TEXT, "")
+        // Check that data was properly recorded.
+        assertTrue(jweMetric.testHasValue())
+        assertEquals(MINIMUM_JWE, jweMetric.testGetCompactRepresentation())
+    }
+
+    @Test
+    fun `disabled JWEs must not record data`() {
+        // Define a 'jweMetric' jwe metric, which will be stored in "store1". It's disabled
+        // so it should not record anything.
+        val jweMetric = JweMetricType(
+            disabled = true,
+            category = "telemetry",
+            lifetime = Lifetime.Application,
+            name = "jwe_metric",
+            sendInPings = listOf("store1")
+        )
+
+        // Attempt to store the JWE.
+        jweMetric.set(HEADER, KEY, INIT_VECTOR, CIPHER_TEXT, AUTH_TAG)
+        // Check that nothing was recorded.
+        assertFalse(
+            "JWEs must not be recorded if they are disabled",
+            jweMetric.testHasValue()
+        )
+    }
+
+    @Test(expected = NullPointerException::class)
+    fun `testGetValue() throws NullPointerException if nothing is stored`() {
+        val jweMetric = JweMetricType(
+            disabled = true,
+            category = "telemetry",
+            lifetime = Lifetime.Application,
+            name = "jwe_metric",
+            sendInPings = listOf("store1")
+        )
+        jweMetric.testGetValue()
```

Comment:
We should also test the result of `testGetValue` (in another test).

Reply:
We already have a test for it on the core: https://github.com/mozilla/glean/blob/main/glean-core/tests/jwe.rs#L92, do you think it's necessary to test again on each binding?